lib/rsolr/client.rb

```diff
@@ -33,6 +32,7 @@ def initialize connection, options = {}
     @update_format = options.delete(:update_format) || RSolr::JSON::Generator
     @update_path = options.fetch(:update_path, 'update')
     @options = options
+    @connection = connection
```

Comment:
I think it is lazy already. I also misunderstood this line, which is why I moved it. This is setting `@connection` from the `connection` argument, not calling the `connection` method. I will move it back as this change should not be needed.